docs/docs/architecture/security.md

```diff
@@ -0,0 +1,97 @@
+Applications interacting with the MarbleRun deployment can decide wether to use the intermediate or the root certificate as CA depending on if they want to notice and handle manifest updates explicitely or not.
+
+For every Marble the Coordinator generates a unique leaf "Marble" certificate and corresponding key pair using ECDSA with P256.
+The Marble certificate is signed by the Marble root certificate.
+The Marble certificate is provisioned to the Marble's enclave via the secure channel established during the [attestation procedure](../features/attestation.md).
+
+![Security architecture](../_media/cert-chain.svg)
+
+
+## Attested TLS (aTLS)
+
+In a CC environment, attested TLS (aTLS) can be used to establish secure connections between two parties using the remote attestation features of the CC components.
+aTLS modifies the TLS handshake by embedding an attestation statement into the TLS certificate. Instead of relying on a certificate authority, aTLS uses this attestation statement to establish trust in the certificate.
+The protocol can be used by clients to verify a server certificate, by a server to verify a client certificate, or for mutual verification (mutual aTLS).
+
+### Mutual TLS authentication
+
+Based on MarbleRun's PKI Marbles are provided with a unique leaf certificate and corresponding private key.
+As described above these are generated by the Coordinator and provided during the Marble's initialization via the secure aTLS channel.
+Depending on the Marble's runtime the certificate can either be used [manually](../workflows/add-service.md#make-your-service-use-the-provided-tls-credentials) or [automatically](../features/transparent-TLS.md) for establishing mutually authenticated TLS connetions.
+
+## Encryption of state
+
+The Coordinator holds the MarbleRuns state, which consists of the [manifest](../features/manifest.md), the [managed secrets](../features/secrets-management.md), and the [certificates for its CA](../features/attestation.md).
+The state is stored encrypted in persistent storage. Therefore Constellation uses [AES128-GCM](https://www.rfc-editor.org/rfc/rfc5116#section-5.1) and a randomly generated 16-byte Data Encryption Key (KEK) to seal the state to persistent storage.
+The DEK is also sealed to persistent storage to recover the state in case of a restart autonomously.
+[SGX's sealing feature](https://www.intel.com/content/www/us/en/developer/articles/technical/introduction-to-intel-sgx-sealing.html) is used for that purpose.
+The Coordinator encrypts the DEK with a key encryption key (KEK).
+The KEK MarbleRun uses the SGX sealing key called `Product key`, which is bound to its `Product ID` and the enclave's author `MRSIGNER` identity.
+In other words, a fresh and benign enclave instance of the same identity can recover that key.
+Hence, as long as the Coordinator is restarted on the same CPU, it can obtain the same KEK from the CPU based on its identity, decrypt the DEK and recover its state.
```

Comment:
consider adding manual recovery (and the involved RSA encryption) here

Reply:
Added